jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -87,16 +87,35 @@ async def job_submission_get(job_submission_id: int = Query(...)):
 )
 async def job_submission_list(
     pagination: Pagination = Depends(),
-    all: Optional[bool] = Query(None),
+    all: Optional[bool] = Query(
+        None, description="If supplied, do not limit job_submissions to only the current user",
+    ),
+    slurm_job_ids: Optional[str] = Query(
+        None,
+        description="Comma-separated list of slurm-job-ids where results must have a matching slurm_job_id",
+    ),
     token_payload: TokenPayload = Depends(guard.lockdown("jobbergate:job-submissions:read")),
 ):
     """
     List job_submissions for the authenticated user.
+
+    :param: all:           Do not limit results to only the current user
+    :param: slurm_job_ids: Limit results job_submissions having one of the supplied slurm_job_ids.
+                           If used, must be a comma separated list of integers.
     """
     identity_claims = IdentityClaims.from_token_payload(token_payload)
     query = job_submissions_table.select()
     if not all:
         query = query.where(job_submissions_table.c.job_submission_owner_email == identity_claims.user_email)
+    if slurm_job_ids is not None:
```

Comment:
If the user sends an empty list, will it work also?

Reply:
Good point. Let me add some logic and unit-tests for that.

Reply:
@heitorPB Addressed in the latest commit.

Reply:
I think an empy list should have the same behaviour as not having the list. Why you decided to return an error to the user?

Reply:
The issue is that we already have a null case for this (when the value is `None`). If we are saying in the docs that the argument should be a comma separated list of ids, then empty string to me seems invalid even if it would work functionally. I made the error condition to remove some ambiguity between `""` and `None`.

Reply:
Slurm does not make this distinction between empty and not defined. I think Jobbergate should follow this strategy. Otherwise it can be confusing to end users.

Reply:
I feel like we're doing a bit of bike-shedding here.

I'm going to remove the extra logic in the implementation from the last commit but still include a unit test to show that empty string has no effect on the filter.

Reply:
Resolved in latest commit.